presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/PlanChecker.java

```diff
@@ -77,26 +91,95 @@ public PlanChecker(FeaturesConfig featuresConfig, boolean forceSingleNode)
 
     public void validateFinalPlan(PlanNode planNode, Session session, Metadata metadata, WarningCollector warningCollector)
     {
+        ensureProvidedInitialized();
         checkers.get(Stage.FINAL).forEach(checker -> checker.validate(planNode, session, metadata, warningCollector));
     }
 
     public void validateIntermediatePlan(PlanNode planNode, Session session, Metadata metadata, WarningCollector warningCollector)
     {
+        ensureProvidedInitialized();
         checkers.get(Stage.INTERMEDIATE).forEach(checker -> checker.validate(planNode, session, metadata, warningCollector));
     }
 
-    public void validatePlanFragment(PlanNode planNode, Session session, Metadata metadata, WarningCollector warningCollector)
+    public void validatePlanFragment(PlanFragment planFragment, Session session, Metadata metadata, WarningCollector warningCollector)
+    {
+        ensureProvidedInitialized();
+        checkers.get(Stage.FRAGMENT).forEach(checker -> checker.validateFragment(planFragment, session, metadata, warningCollector));
+    }
+
+    private static List<Checker> fromSpi(List<com.facebook.presto.spi.plan.PlanChecker> checkers)
+    {
+        return checkers.stream().map(SpiCheckerAdapter::new).collect(Collectors.toList());
+    }
+
+    private void ensureProvidedInitialized()
     {
-        checkers.get(Stage.FRAGMENT).forEach(checker -> checker.validate(planNode, session, metadata, warningCollector));
+        if (!providedInitialized) {
+            if (providerMgr != null) {
```

Comment:
ProviderManager should never be allowed to be null, and then you won't need this check.